playground/backend/internal/storage/cloud_storage.go

```diff
@@ -0,0 +1,218 @@
+// --------/Example1
+// --------/Example2
+// ----  ...
+// ...
+type CloudStorage struct {
+}
+
+func New() *CloudStorage {
+	return &CloudStorage{}
+}
+
+// GetExample returns the source code of the example
+func (cd *CloudStorage) GetExample(ctx context.Context, examplePath string) (*string, error) {
+	extension := getFileExtensionBySdk(examplePath)
+	data, err := cd.getFileFromStorage(ctx, examplePath, extension)
+	if err != nil {
+		return nil, err
+	}
+	result := string(data)
+	return &result, nil
+}
+
+// GetExampleOutput returns the run output of the example
+func (cd *CloudStorage) GetExampleOutput(ctx context.Context, examplePath string) (*string, error) {
+	data, err := cd.getFileFromStorage(ctx, examplePath, OutputExtension)
+	if err != nil {
+		return nil, err
+	}
+	result := string(data)
+	return &result, nil
+}
+
+// GetListOfExamples returns the list of stored example at cloud storage bucket
+func (cd *CloudStorage) GetListOfExamples(ctx context.Context, sdk pb.Sdk, category string) (*Examples, error) {
+	client, err := storage.NewClient(ctx, option.WithoutAuthentication())
+	if err != nil {
+		return nil, fmt.Errorf("storage.NewClient: %v", err)
+	}
+	defer client.Close()
+
+	ctx, cancel := context.WithTimeout(ctx, time.Second*Timeout)
+	defer cancel()
+
+	bucket := client.Bucket(BucketName)
+	prefix := sdk.String()
+	if sdk == pb.Sdk_SDK_UNSPECIFIED {
+		prefix = ""
+	}
+	examples := make(Examples, 0)
+	it := bucket.Objects(ctx, &storage.Query{
+		Prefix: filepath.Join(prefix, category),
+	})
+	for {
+		attrs, err := it.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			return nil, fmt.Errorf("Bucket(%q).Objects: %v", BucketName, err)
+		}
+		path := attrs.Name
+		if isFullPathToExample(path) {
+			infoPath := filepath.Join(path, MetaInfoName)
+			rc, err := bucket.Object(infoPath).NewReader(ctx)
+			if err != nil {
+				logger.Errorf("Object(%q).NewReader: %v", infoPath, err.Error())
+				continue
+			}
+			data, err := ioutil.ReadAll(rc)
+			if err != nil {
+				logger.Errorf("ioutil.ReadAll: %v", err.Error())
+				continue
+			}
+			exampleInfo := pb.Example{}
+			err = json.Unmarshal(data, &exampleInfo)
+			if err != nil {
+				logger.Errorf("json.Unmarshal: %v", err.Error())
+				continue
+			}
+
+			exampleInfo.CloudPath = path
+			exampleInfo.Name = filepath.Base(path)
+			cd.writeExample(path, &examples, &exampleInfo)
+			rc.Close()
+		}
+	}
+	return &examples, nil
+}
+
+func isFullPathToExample(path string) bool {
+	return strings.Count(path, string(os.PathSeparator)) == 3 && path[len(path)-1] == os.PathSeparator
+}
+
+func (cd *CloudStorage) writeExample(path string, examples *Examples, exampleInfo *pb.Example) {
+	splintedPath := strings.Split(path, string(os.PathSeparator))
+	sdk := splintedPath[0]      // the path of the form "sdk/category/example", where the first part is sdk
+	category := splintedPath[1] // and the second part is the name of the category
+	catMap, ok := (*examples)[sdk]
```

Comment:
catMap -> categoriesWithExamples. Please don't use type in variable name

Reply:
I double-checked it, and yes, type in a variable it's pretty bad practice. It's ok for go to have short names for not very important variables or list if it is temp list, however, put type in the name - bad practice.

Reply:
I forgot to rename it, it was developing name when I nothing had come to my mind at the moment